File: SEM5-PI-WEBAPI/Domain/Dock/DockMapper.cs
using SEM5_PI_WEBAPI.Domain.Dock.DTOs;

namespace SEM5_PI_WEBAPI.Domain.Dock;

public class DockMapper
{
    public static DockDto RegisterDockDto(EntityDock instance)
    {
        return new DockDto(
            instance.Id.AsGuid(),
            instance.Code,
            instance.PhysicalResourceCodes,
            instance.Location,
            instance.LengthM,
            instance.DepthM,
            instance.MaxDraftM,
            instance.Status,
            instance.AllowedVesselTypeIds
        );
    }
}

File: client-ui/src/features/docks/mappers/dockMapper.ts
import type { DockDto } from "../dto/dockDtos";
import type { Dock } from "../domain/dock";

export function mapDockDto(dto: DockDto): Dock {
    return {
        id: dto.id,
        code: dto.code,
        physicalResourceCodes: dto.physicalResourceCodes,
        location: dto.location,
        lengthM: dto.lengthM,
        depthM: dto.depthM,
        maxDraftM: dto.maxDraftM,
        status: dto.status,
        allowedVesselTypeIds: dto.allowedVesselTypeIds,
    };
}
